pkg/server/server.go

```diff
@@ -595,25 +651,9 @@ func (s *ProxyServer) serveRecvFrontend(frontend *GrpcFrontend, recvCh <-chan *c
 	}
 
 	// TODO: Log an error if the frontend stream is closed with an established tunnel still open.
-	klog.V(5).InfoS("Close streaming", "connectionID", firstConnID)
-
+	klog.V(5).InfoS("Close frontend streaming", "connectionID", firstConnID)
 	if backend == nil {
 		klog.V(2).InfoS("Streaming closed before backend intialized")
-	} else if firstConnID != 0 {
-		pkt := &client.Packet{
-			Type: client.PacketType_CLOSE_REQ,
-			Payload: &client.Packet_CloseRequest{
-				CloseRequest: &client.CloseRequest{
-					ConnectID: firstConnID,
-				},
-			},
-		}
-		if err := backend.Send(pkt); err != nil {
-			// TODO: Determine whether this is expected, and log at error level if appropriate.
-			// If that connection shutdown was innitiated by the backend, this is expected.
-			// If initiated by the frontend, it is not.
-			klog.V(2).InfoS("CLOSE_REQ to Backend failed", "error", err)
-		}
```

Comment:
Why delete this? Is the idea just to reduce the places we close the backend by deferring the close to after the FE stream is closed?

Reply:
Avoid duplication (both server code and duplicate close messages).

This handling is moved from the bottom of `serveRecvFrontend` (just before exit) to a `Proxy` defer, which isn't very far away.